cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -1286,15 +1405,18 @@ static void _umm_free( void *ptr ) {
    *        on the free list!
    */
 
-  /* Protect the critical section... */
-  UMM_CRITICAL_ENTRY();
-
   /* Figure out which block we're in. Note the use of truncated division... */
 
   c = (((char *)ptr)-(char *)(&(umm_heap[0])))/sizeof(umm_block);
 
+  /* Protect the critical section... */
```

Comment:
Maybe move this down to 1416 since the debug print isn't critical?

Reply:
I agree.